### PR TITLE
citations: fix counting twice same reference

### LINKED ISSUE #662: Citation count mismatch
The following record in the demo records shows a citation count of 4.

screen shot 2015-12-15 at 10 49 15 am

But after clicking on the link a search `refersto:1295494` is performed and the results returned are 3.

This mismatch needs to be checked.

### PR DESCRIPTION
**Pull request by tomaszgy**
* closes #662

Signed-off-by: Tomasz <tomaszgy@gmail.com>

### CODE REVIEW

**Comment by kaplun** [comment]
Can you describe the fix in the commit log?

File: inspirehep/modules/migrator/tasks.py [hunk 1]
```diff
@@ -228,11 +229,15 @@ def get_records_to_update_generator(citation_lookup):
 
         # update lookup dictionary based on references of the record
         if 'references' in record['_source']:
+            unique_refs_ids = Set()
             references = record['_source']['references']
             for reference in references:
                 recid = reference.get('recid')
                 if recid:
-                    citations_lookup[unicode(recid)] += 1
+                    unique_refs_ids.add(unicode(recid))
```
Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: Why are you casting the recids to unicode? They should be simple integers.

File: inspirehep/modules/migrator/tasks.py [hunk 2]
```diff
@@ -228,11 +229,15 @@ def get_records_to_update_generator(citation_lookup):
 
         # update lookup dictionary based on references of the record
         if 'references' in record['_source']:
+            unique_refs_ids = Set()
```
Comment (jmartinm) [inline inspirehep/modules/migrator/tasks.py]: Is there any difference with using the built-in keyword `set()` ?

File: inspirehep/modules/migrator/tasks.py [hunk 3]
```diff
@@ -198,6 +198,7 @@ def add_citation_counts():
     from elasticsearch.helpers import bulk as es_bulk
     from elasticsearch.helpers import scan as es_scan
     from collections import Counter
+    from sets import Set
```
Comment (kaplun) [inline inspirehep/modules/migrator/tasks.py]: In modern Python you can use the builtin `set` type, which is faster than the pythonic `sets.Set`

**Comment by jmartinm** [comment]
For the commit log message, please start with capital letter and end with a dot:

```
* Closes #662.
```

**Comment by tomaszgy** [comment]
Sure. I'll do it. Also as a supplement, I list below the records in the data set that have the "doubled references":

Record 1398152, doubled reference: 593382
Record 1377199, doubled reference: 103744
Record 1377199, doubled reference: 712925
Record 1377199, doubled reference: 1295494
Record 1398153, doubled reference: 1239348
Record 1400799, doubled reference: 1315584

File: inspirehep/modules/migrator/tasks.py (hunk 3 above)
Comment (tomaszgy) [inline inspirehep/modules/migrator/tasks.py]: Wow, so many comments in the meantime :smile: I'll update the commit.

**Comment by tomaszgy** [comment]
Updated.

**Comment by jmartinm** [comment]
Looks good, sorry one small commit log change. Can you keep the maximum line characters to 80 in the commit log message? Check `git log` to see examples.

**Comment by tomaszgy** [comment]
Corrected.

**Pull request merged**